pkg/util/bandwidth/linux.go

```diff
@@ -153,13 +153,20 @@ func (t *tcShaper) findCIDRClass(cidr string) (classAndHandleList [][]string, fo
 			continue
 		}
 		if strings.Contains(line, spec) {
+			var resultTmp []string
+
 			parts := strings.Split(filter, " ")
-			// expected tc line:
-			// filter parent 1: protocol ip pref 1 u32 fh 800::800 order 2048 key ht 800 bkt 0 flowid 1:1
-			if len(parts) != 19 {
+			if len(parts) == 19 {
+				// expected tc line:
+				// filter parent 1: protocol ip pref 1 u32 fh 800::800 order 2048 key ht 800 bkt 0 flowid 1:1
+				resultTmp = []string{parts[18], parts[9]}
+			} else if len(parts) == 21 {
```

Comment:
This is just going to fail the next time they add something. The output is clearly in the form of key/value pairs. We should find the keys we want and output the corresponding values.

(likewise in `nextClassID` at least)

Reply:
yes, I agree with @danwinship . So I use regexp to extract the info we want from the output. This is the [fix PR](https://github.com/kubernetes/kubernetes/pull/83572) and I hope it helps.